### ISSUE TITLE
[Issue] AGS indicators widget leaves a gray line at the top of the screen

### ISSUE THREAD

**Issue by ricky8k** [end-4/dots-hyprland#268]
- **I have read the [wiki](https://end-4.github.io/dots-hyprland-wiki/en/i-i/03troubleshooting)**
- Linux distro: EndeavourOS (Arch)

---
While AGS is running, a faint horizontal gray line can be seen near the top-middle of the screen.

It seems to be the indicators widget causing the issue, as this goes away when `forMonitors(Indicator)` is removed from `~/.config/ags/config.js`. Killing AGS will have the same effect.

I've attached a video demonstrating the gray line in front of a white video frame and it disappearing when commenting this line.

https://github.com/end-4/dots-hyprland/assets/92121005/fdcf2810-7ea9-4da2-9d6d-f38a0c07f2fc

Edit: Better demonstration in https://github.com/end-4/dots-hyprland/issues/268#issuecomment-1949636705

**Comment by clsty** [comment]
To be honest, I can't see any faint gray line in your video both before and after commenting that line.

Don't get me wrong, I'm not saying you were lying or something.
Maybe some other thing is wrong.

**Comment by clsty** [comment]
Can you try to play this video on another device, such as a mobile phone, and see if there's still a faint line?

**Comment by ricky8k** [comment]
> Can you try to play this video on another device, such as a mobile phone, and see if there's still a faint line?

Yes, I can still see the faint line in my video from my phone and other computer. Are you able to see it in the video?

**Comment by ricky8k** [comment]
I recorded the same gray line on a white video frame, this time by restarting AGS a few times. You can see when it protrudes for a moment when I restart it.

https://github.com/end-4/dots-hyprland/assets/92121005/469c0ca4-7c9a-4f59-9cb5-90f3639bf359

**Comment by clsty** [comment]
> Are you able to see it in the video?

Nope.

> I'm not sure how I can make it easier to spot in the video...

I see your mouse cursor. You've already spot where the line is with the cursor, didn't you?

---

This could be a really weird problem.

Possibilities:
- codec of video
- HDR and SDR thingy
- 10-bit and 8-bit color thingy

(I know little on any one above though)

**Comment by clsty** [comment]
> I recorded the same gray line on a white video frame, this time by restarting AGS a few times. You can see when it protrudes for a moment when I restart it.
> recording_20240216_23-03-57.mp4

Oh, I see it this time, a short broken black line in the center of screen.

**Comment by ricky8k** [comment]
> > Are you able to see it in the video?
> 
> Nope.
> 
> > I'm not sure how I can make it easier to spot in the video...
> 
> I see your mouse cursor. You've already spot where the line is with the cursor, didn't you?
> 
> This could be a really weird problem.
> 
> Possibilities:
> 
>     * codec of video
> 
>     * HDR and SDR thingy
> 
>     * 10-bit and 8-bit color thingy
> 
> 
> (I know little on any one above though)

I don't think the issue is the video, since it seems to affect "system-wide" (AGS overlays everything) on my end. I used the video as an example, but you could spot it in photos, applications, etc. if it is a light solid color.

**Comment by clsty** [comment]
> I don't think the issue is the video, since it seems to affect "system-wide" (AGS overlays everything) on my end. I used the video as an example, but you could spot it in photos, applications, etc. if it is a light solid color.

Sure I know it's system-wide. I talk about video because you can see it in your video, while I can't.
i.e. There're different devices playing the same video but with different looks.

**Comment by ricky8k** [comment]
My apologies, I thought you meant the white video I used as an example.

**Comment by clsty** [comment]
![screenshot](https://github.com/end-4/dots-hyprland/assets/129247596/a62077a2-a498-4a76-9919-b7c225f54b0b)

This is what I see.

Actually there's still no faint gray line, but two short black lines. I guess they're different things (?)

**Comment by ricky8k** [comment]
Yeah, this should be the line. In the screenshot you sent is when the line protrudes the most before going back to the faint line.

**Comment by end-4** [comment]
Known problem. This is because those widgets are not hidden in a regular Revealer, but a custom one that allows custom easing. The movement looks nicer, and that line only appears for a very short period so I just ignored it
